subsys/net/lib/lwm2m/lwm2m_obj_firmware_pull.c

```diff
@@ -27,9 +27,10 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #define NETWORK_CONNECT_TIMEOUT	K_SECONDS(10)
 #define PACKET_TRANSFER_RETRY_MAX	3
 
-static char firmware_uri[URI_LEN];
+static char firmware_uri[CONFIG_LWM2M_FIRMWARE_INSTANCE_COUNT][URI_LEN];
```

Comment:
Genrally, changes in this file will be in conflict with https://github.com/zephyrproject-rtos/zephyr/pull/40552, which I believe will be integrated in NCS on the next upmerge (it's near finalization). I think we should rather build our solution for multiple FW instances on top of that PR (specifically the reusable pull context).

Reply:
Yeah, I agree. As that PR has now been merged to the upstream-zephyr, I will do some rework over that and make a PR to the upstream.